src/main/scala/com/fulcrumgenomics/umi/ReviewConsensusVariants.scala

```diff
@@ -0,0 +1,139 @@
+  @arg(flag="g", doc="BAM file of grouped raw reads used to build consensuses.") val groupedBam : PathToBam,
+  @arg(flag="r", doc="Reference fasta file.") val ref: PathToFasta,
+  @arg(flag="o", doc="Basename of output files to create.") val output : PathPrefix,
+  @arg(flag="t", doc="The SAM/BAM tag used to uniquely identify source molecules.") val tag: String = "MI"
+)extends FgBioTool {
+  Io.assertReadable(Seq(input, consensusBam, groupedBam, ref))
+  Io.assertCanWriteFile(output)
+  private val refFile = ReferenceSequenceFileFactory.getReferenceSequenceFile(ref)
+  private val dict = refFile.getSequenceDictionary
+
+  private[umi] case class LocAndRef(chrom: String, start: Int, refBase: Char) {
+    def toQueryInterval: QueryInterval = new QueryInterval(dict.getSequenceIndex(chrom), start, start)
+  }
+
+  override def execute(): Unit = {
+    def f(ext: String): Path = output.getParent.resolve(output.getFileName + ext)
+    val consensusIn = SamReaderFactory.make().open(consensusBam)
+    val groupedIn   = SamReaderFactory.make().open(groupedBam)
+    val factory = new SAMFileWriterFactory().setCreateIndex(true)
+    val consensusOut = factory.makeWriter(consensusIn.getFileHeader, true, f(".consensus.bam").toFile, ref.toFile)
+    val groupedOut   = factory.makeWriter(groupedIn.getFileHeader, true, f(".grouped.bam").toFile, ref.toFile)
+
+    // Pull out the consensus reads
+    val variants        = loadPositions(path=input, refFile=refFile)
+    val variantsByChrom = variants.groupBy(v => v.chrom)
+    val queries         = variants.map(_.toQueryInterval).toArray
+    val sources         = mutable.HashSet[String]()
+    consensusIn.query(queries, false).filter(r => nonReferenceAtAnyVariant(r, variantsByChrom)).foreach { rec =>
+      consensusOut.addAlignment(rec)
+      Option(rec.getStringAttribute(tag)).map(mi => mi.takeWhile(_ != '/')).foreach(sources.add)
+    }
+    consensusIn.safelyClose()
+    consensusOut.close()
+
+    // And now the raw reads
+    groupedIn.query(queries, false)
+      .filter(rec => Option(rec.getStringAttribute(tag)).map(mi => mi.takeWhile(_ != '/')).exists(sources.contains))
+      .foreach(groupedOut.addAlignment)
+    groupedIn.safelyClose()
+    groupedOut.close()
+  }
+
+  /** Loads the set of locations for review. */
+  private[umi] def loadPositions(path: FilePath, refFile: ReferenceSequenceFile): Seq[LocAndRef] = {
+    val buffer = ListBuffer[LocAndRef]()
+
+    if (VcfExtensions.exists(ext => path.getFileName.toString.endsWith(ext))) {
+      val in = new VCFFileReader(input.toFile, false)
+      in.filter(_.isSNP).foreach(v => buffer += LocAndRef(v.getContig, v.getStart, v.getReference.getBases()(0).toChar.toUpper))
+    }
+    else {
+      val list = IntervalList.fromFile(path.toFile).uniqued(false)
+      for (i <- list; pos <- i.getStart to i.getEnd) {
+        buffer += LocAndRef(i.getContig, pos, refFile.getSubsequenceAt(i.getContig, pos, pos).getBases()(0).toChar.toUpper)
```

Comment:
`getSubsequenceAt` is likely slow, how about `refFile.getSequence(i.getContig).getBases()(pos-1).toChar.toUpper`?